x-pack/plugin/ql/src/main/java/org/elasticsearch/xpack/ql/util/SpatialCoordinateTypes.java

```diff
@@ -108,7 +108,19 @@ public BytesRef wktToWkb(String wkt) {
         }
     }
 
+    public Geometry wktToGeometry(String wkt) {
+        try {
+            return WellKnownText.fromWKT(GeometryValidator.NOOP, false, wkt);
+        } catch (Exception e) {
+            throw new IllegalArgumentException("Failed to parse WKT: " + e.getMessage(), e);
```

Comment:
Use InvalidArgumentException instead (through-out the code as well).

Reply:
Interesting. I took a look around, and found many more usages of IllegalArgumentException in the csv-spec tests and related functions. Looks like InvalidArgumentException is used only in a few places. Are we planning to change this all over to InvalidArgumentException? Perhaps that should be done as a separate PR for all of these cases, since it is much wider scope that this PR?